app/modules/mqtt.c

```diff
@@ -873,8 +885,7 @@ static sint8 socket_connect(struct espconn *pesp_conn)
 #ifdef CLIENT_SSL_ENABLE
   if(mud->secure)
   {
-      espconn_secure_set_size(ESPCONN_CLIENT, 5120); /* set SSL buffer size */
```

Comment:
You remove `espconn_secure_set_size()` here and perform it once during module initialization. I suspect that this  creates interdependencies to other modules that change the setting in the meantime. You can't be sure about the current value when execution reaches this point unless there's a global convention.